### PR TITLE
Fix import for completions

### PR DESCRIPTION
**Pull request by bpinto**

### CODE REVIEW

**Comment by bpinto** [comment]
@bucaran see how these tests fail

**Comment by bpinto** [comment]
Can we solve this without changing `_prepend_tree` to ignore completions folder?

**Comment by bucaran** [comment]
If this tests should pass, merge, and then we can fix import.

**Comment by bpinto** [comment]
Tests do not pass.

Expectation:
it_does_not_import_completion_files
it_adds_completion_files_to_completion_path

**Comment by bpinto** [comment]
Currently we import files under completion folder by adding them to fish_function_path.

**Comment by bucaran** [comment]
I see, so we should not import files inside `completions`?

So, how do completions work then? Does fish check any directories named `completions` and loads them automatically or something?

**Comment by bpinto** [comment]
AFAIK, it is similar to how function works, if you want the completions for a command called theme, it will look for a `theme.fish` file on `fish_complete_path` env var.

**Comment by bpinto** [comment]
Do you have a better solution to this problem then the one on the last commit?

File: functions/_prepend_tree.fish [hunk 1]
```diff
@@ -84,13 +84,9 @@ function _prepend_tree -d "Add a dependency tree to the Fish path."
...
 
-  # Null wildcard expansion will break the for loop even if $path is valid.
-  # $subs will become an empty list for directories without sub directories
-  # which is safe to use in the loop.
-  set -l subs $path/**/
-
-  # Traverse $path and $subs prepending only directories with matches.
-  for dir in $path $subs
+  # Traverse $path prepending only directories with matches. Excludes completions folder.
+  test -d $path
```
Comment (bucaran) [inline functions/_prepend_tree.fish]: Why do we need to exclude completion folders?

Comment (bucaran) [inline functions/_prepend_tree.fish]: Nevermind. All is clear now.

**Comment by bucaran** [comment]
Another feature of `fish` I didn't understand clearly. This is an awesome feature by the way.

**Comment by bpinto** [comment]
Fish is full of hidden gems. :+1:

**Comment by bucaran** [comment]
The solutions does seem a bit hacky, maybe we can come up with something cleaner? Let me see.

**Comment by bpinto** [comment]
You only say that now? After I merged? :(

I hope you find something cleaner.

**Comment by bucaran** [comment]
Sorry, it took my brain thousands of nanoseconds to finally grasp fish completions.

**Comment by bucaran** [comment]
To recapitulate:

> AFAIK, it is similar to how function works, if you want the completions for a command called theme, it will look for a theme.fish file on fish_complete_path env var.

:+1:

**Comment by bpinto** [comment]
:+1:

**Pull request merged**